Allow photochemistry schemes to have parameters at different times.

DWesl opened the pull request:
In particular, allow the ozone scheme to have two extra sets of parameters during the ozone hole without crashing the model.

An update to #223/#213 to keep the model from crashing after NOAA-EMC/global-workflow#3386

DWesl committed:
BUG: Allow photochemistry schemes to have parameters at different times.

In particular, allow the ozone scheme to have two extra sets of parameters during the ozone hole without crashing the model.

DWesl committed:
Merge branch 'ufs/dev' of ufs-community/ccpp-physics into photochemistry-index-fix

XiaqiongZhou-NOAA commented on physics/Interstitials/UFS_SCM_NEPTUNE/GFS_phys_time_vary.scm.F90:
```diff
@@ -769,21 +769,15 @@ subroutine GFS_phys_time_vary_timestep_init (
             call ozphys%update_o3prog(jindx1_o3, jindx2_o3, ddy_o3, rjday, n1, n2, ozpl)
          endif
 
+         call find_photochemistry_index(h2ophys%ntime, h2ophys%time, rjday, n1, n2)
```
this line should move inside if (h2o_phys) then?

DWesl replied:
I moved all calls inside the relevant conditional, does that work?

DWesl committed:
STY: Move the find_photochemistry_index calls inside the conditionals.

With a separate index for each photochemistry scheme, we can move the index-finding calls
in the conditional for the photochemistry scheme.

DWesl commented:
Merged into #265
